### PR TITLE
Document launching child proc using dotnet-trace and dotnet-counters

### PR DESCRIPTION
**Pull request by sywhang**
## Summary

This documents a new dotnet-trace and dotnet-counters feature where an application can be launched and monitored/traced from its startup, introduced by https://github.com/dotnet/diagnostics/pull/1635. 

This PR can be merged once we officially ship the new versions of the diagnostics CLI tools (which is scheduled to be some time next week or so).

cc @tommcdon

### CODE REVIEW

File: docs/core/diagnostics/dotnet-counters.md
Comment (hoyosjs) [inline docs/core/diagnostics/dotnet-counters.md:81]:
```suggestion
  After the collection configuration parameters, the user can append `--` followed by a command to start a .NET application with at least a 5.0 runtime. `dotnet-counters` will launch a process with the provided command and collect the requested metrics. This is often useful to collect metrics for the application's startup path and can be used to diagnose or monitor issues that happen early before or shortly after the main entrypoint.
```

File: docs/core/diagnostics/dotnet-counters.md
Comment (hoyosjs) [inline docs/core/diagnostics/dotnet-counters.md:249]:
```suggestion
  > dotnet-counters monitor --counters System.Runtime[working-set,gc-heap-size] -- my-aspnet-server.exe arg1 arg2
```

File: docs/core/diagnostics/dotnet-trace.md
Comment (hoyosjs) [inline docs/core/diagnostics/dotnet-trace.md:190]: This disclaimer should either be also in .NET counters or just the mention in the parameter documentation. Given how common it is to just look at the examples, I'd say the first option is better..

File: docs/core/diagnostics/dotnet-trace.md
Comment (hoyosjs) [inline docs/core/diagnostics/dotnet-trace.md:219]:
```suggestion
You can stop collecting the trace by pressing `<Enter>` or `<Ctrl + C>` key. Doing this will also exit `hello.exe`.
```

File: docs/core/diagnostics/dotnet-counters.md
Comment (hoyosjs) [inline docs/core/diagnostics/dotnet-counters.md:175]: Hm. Not sure if prefixing is the right word here. Probably we should be consistent in the places where we document this.

File: docs/core/diagnostics/dotnet-trace.md
Comment (hoyosjs) [inline docs/core/diagnostics/dotnet-trace.md:199]: A sample with parameters here would be good too, just as the one in counters.

File: docs/core/diagnostics/dotnet-counters.md
Comment (sywhang) [reply docs/core/diagnostics/dotnet-counters.md:69]: Nope, this is new. The old syntax is still supported, but I'm removing it from the docs in favor of the new syntax.

Comment (hoyosjs) [reply docs/core/diagnostics/dotnet-counters.md:69]: Have these been released? Otherwise we'd end up with customers having tools that don't support this.

Comment (sywhang) [reply docs/core/diagnostics/dotnet-counters.md:69]: Not yet - which is why I mentioned in the PR description that we should only merge this after the tools have been released :)

Comment (hoyosjs) [reply docs/core/diagnostics/dotnet-counters.md:69]: Ah. The diff viewer didn't list the PR description. That's my bad for not reading.

File: docs/core/diagnostics/dotnet-trace.md
Comment (hoyosjs) [inline docs/core/diagnostics/dotnet-trace.md:112]: Maybe copy the same blurb from counters here as well?

File: docs/core/diagnostics/dotnet-counters.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-counters.md:54]:
```suggestion
dotnet-counters collect [-h|--help] [-p|--process-id] [--refreshInterval] [--counters] [--format] [-o|--output] [-- <command>]
```
I suggest we document it as if --counters was required to specify the list. If people don't know about the secret back-compatible syntax I don't think they benefit from learning it.

File: docs/core/diagnostics/dotnet-counters.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-counters.md:67]:
```suggestion
- **`--counters <COUNTERS>`**
```

File: docs/core/diagnostics/dotnet-counters.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-counters.md:159]:
```suggestion
dotnet-counters monitor [-h|--help] [-p|--process-id] [--refreshInterval] [--counters] [-- <command>]
```

File: docs/core/diagnostics/dotnet-counters.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-counters.md:172]:
```suggestion
- **`--counters <COUNTERS>`**
```

File: docs/core/diagnostics/dotnet-trace.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-trace.md:197]: I would delete lines 197 and 199 and move the explanation on line 205 up here to be more succinct. 205 is the better of the two explanations.

File: docs/core/diagnostics/dotnet-trace.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-trace.md:205]: People would probably figure it out but I suggest adding similar text to what the previous examples used to make it unambiguous that this is the output of the command, not the command itself.

 "The preceding command generates output similar to the following:"

**Review by noahfalk — APPROVED** [review]
Great to see this 👍 A few minor suggestions inline.

File: docs/core/diagnostics/dotnet-counters.md
Comment (sdmaclea) [reply docs/core/diagnostics/dotnet-counters.md:159]:
```suggestion
dotnet-counters monitor [-h|--help] [-p|--process-id] [--refreshInterval] [--counters <counters-list>] [-- <command>]
```

File: docs/core/diagnostics/dotnet-counters.md
Comment (sdmaclea) [reply docs/core/diagnostics/dotnet-counters.md:54]:
```suggestion
dotnet-counters collect [-h|--help] [-p|--process-id] [--refreshInterval] [--counters <COUNTERS>] [--format] [-o|--output] [-- <command>]
```

File: docs/core/diagnostics/dotnet-counters.md
Comment (sdmaclea) [inline docs/core/diagnostics/dotnet-counters.md:188]: If we are going to recommend using the `--counters <COUNTERS>` convention, we should update the examples too.

**Review by sdmaclea — APPROVED** [review]
Probably still needs a review form a document style and consistency perspective too.  From docs maintainers....

**Comment by shirhatti** [comment]
I'd like to see a blurb on process lifetime address in the docs. I had a few questions that weren't answered after reading the docs:
- What happens on ctrl+c/SIGTERM to the child process being traced?
- Do we propagate signals to the child process? 
- What happens to collection when the child process exits?

**Review by shirhatti — APPROVED** [review]

File: docs/core/diagnostics/dotnet-trace.md
Comment (hoyosjs) [inline docs/core/diagnostics/dotnet-trace.md:112]:
```suggestion
  After the collection configuration parameters, the user can append `--` followed by a command to start a .NET application with at least a 5.0 runtime. This may be helpful when diagnosing issues that happen early in the process, such as startup performance issue or assembly loader and binder errors.
```

**Review by hoyosjs — APPROVED** [review]

**Pull request merged**
